sdk/src/logging/FileLogDestination.cpp

```diff
@@ -70,38 +111,160 @@ const std::string& FileLogOptions::getFileMode() const
 }
 
+int FileLogOptions::getRotationDays() const
+{
+   return m_rotationDays;
+}
+
 bool FileLogOptions::doRotation() const
 {
    return m_doRotation;
 }
 
+bool FileLogOptions::warnSyslog() const
+{
+   return m_warnSyslog;
+}
+
 bool FileLogOptions::includePid() const
 {
    return m_includePid;
 }
 
+void FileLogOptions::setDeletionDays(int in_deletionDays)
+{
+   m_deletionDays = in_deletionDays;
+}
+
+void FileLogOptions::setDirectory(const FilePath& in_directory)
+{
+   m_directory = in_directory;
+}
+
+void FileLogOptions::setFileMode(const std::string& in_fileMode)
+{
+   m_fileMode = in_fileMode;
+}
+
+void FileLogOptions::setForceDirectory(bool in_forceDirectory)
+{
+   m_forceDirectory = in_forceDirectory;
+}
+
+void FileLogOptions::setMaxRotations(int in_maxRotations)
+{
+   m_maxRotations = in_maxRotations;
+}
+
+void FileLogOptions::setMaxSizeMb(double in_maxSizeMb)
+{
+   m_maxSizeMb = in_maxSizeMb;
+}
+
+void FileLogOptions::setDoRotation(bool in_doRotation)
+{
+   m_doRotation = in_doRotation;
+}
+
+void FileLogOptions::setRotationDays(int in_rotationDays)
+{
+   m_rotationDays = in_rotationDays;
+}
+
+void FileLogOptions::setWarnSyslog(bool in_warnSyslog)
+{
+   m_warnSyslog = in_warnSyslog;
+}
+
 // FileLogDestination ==================================================================================================
 struct FileLogDestination::Impl
 {
-   Impl(unsigned int in_id, const std::string& in_name, FileLogOptions in_options) :
+   Impl(const std::string& in_name, FileLogOptions in_options) :
       LogOptions(std::move(in_options)),
-      LogName(in_name + ".log"),
-      RotatedLogName(in_name + ".old.log"),
-      Id(in_id)
+      LogName(in_name + ".log")
    {
-      LogOptions.getDirectory().ensureDirectory();
-   };
+      if (!LogOptions.getDirectory().exists())
+      {
+         // Create each directory listed in the final desired logging directory
+         // with 0775 permissions - this will ensure that regular users cannot
+         // access/modify the directory/contents and requires other RStudio products
+         // to create their subdirectory as root
+         std::string pathStr = LogOptions.getDirectory().getAbsolutePath();
+         std::vector<std::string> pathParts;
+         boost::split(pathParts, pathStr, boost::is_any_of("/"));
+
+         FilePath path("/");
+         for (const auto& part : pathParts)
+         {
+            if (part.empty())
+               continue;
+
+            Error error = path.completeChildPath(part, path);
+            if (error)
+               return;
+
+            if (!path.exists())
+            {
+               Error error = path.ensureDirectory();
+               if (error)
+                  return;
+
+               path.changeFileMode(FileMode::USER_READ_WRITE_EXECUTE_GROUP_READ_WRITE_EXECUTE_ALL_READ_EXECUTE);
+            }
+         }
+      }
+      else if (LogOptions.getDirectory().getAbsolutePath() == "/var/log/rstudio/rstudio-server")
```

Comment:
Let's filter out this entire `else if` block because this is bug-fix code to clean up some log directories for RStudio Server / Launcher, but not applicable here in the SDK code.

Reply:
Looks like this is still here, but you resolved. Is this not something we can filter out?